app/main.py

```diff
@@ -1,2 +1,334 @@
+        self._values = dictionary_values.__dict__["_values"]
+        self._index = 0
+
+    def __iter__(self) -> Self:
+        return self
+
+    def __next__(self) -> _TV:
+        if self._index == len(self._values):
+            raise StopIteration
+
+        key = self._values[self._index]
+        self._index += 1
+        return key
+
+
+class DictionaryItemsIterator[_TK, _TV]:
+
+    def __init__(self, dictionary_items: DictionaryItems[_TK, _TV], /) -> None:
+        self._items = dictionary_items.__dict__["_items"]
+        self._index = 0
+
+    def __iter__(self) -> Self:
+        return self
+
+    def __next__(self) -> tuple[_TK, _TV]:
+        if self._index == len(self._items):
+            raise StopIteration
+
+        key, value = self._items[self._index]
+        self._index += 1
+        return key, value
+
+
+class Dictionary[_TK, _TV](Mapping):
+
+    def __init__(self) -> None:
+        self._hash_table: list[None | Node] = [None] * 8
+        self._length = 0
+
+    def __str__(self) -> str:
+        return (
+            "Dictionary({"
+            + ", ".join(
+                f"{repr(node.key)}: {repr(node.value)}"
+                for node in self._hash_table if node is not None
+            ) + "})"
+        )
+
+    def __iter__(self) -> DictionaryKeysIterator[_TK]:
+        return iter(self.keys())
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __getitem__(self, key: _TK, /) -> _TV:
+        return self._manage_hash_table(key, get_item=True)
+
+    def __setitem__(self, key: _TK, value: _TV, /) -> None:
+        self._manage_hash_table(key, value, set_item=True)
+
+    def __delitem__(self, key: _TK, /) -> None:
+        # hash_key = hash(key)
+        # index = hash_key % len(self._hash_table)
+        #
+        # if self._key_by_index_is_found(index, key):
+        #     self._hash_table[index] = None
+        #     self._length -= 1
+        #
+        # else:
+        #     index2 = index + 1
+        #     while index2 != index:
+        #         if index2 >= len(self._hash_table):
+        #             index2 = 0
+        #
+        #         if self._key_by_index_is_found(index2, key):
+        #             self._hash_table[index2] = None
+        #             self._length -= 1
+        #             break
+        #
+        #         if index2 != index:
+        #             index2 += 1
+        #     else:
+        #         raise KeyError(f"Key '{key}' is not in dictionary")
+        self._manage_hash_table(key, delete_item=True)
+
+    def _get_item_by(self, index: int, /) -> _TV:
+        return self._hash_table[index].value
+
+    def _set_item_by(
+        self,
+        index: int,
+        key: _TK,
+        value: _TV,
+        /,
+        *,
+        key_exist: bool = False
+    ) -> None:
+        if key_exist:
+            self._hash_table[index].value = value
+        else:
+            self._hash_table[index] = Node(key, value)
```

Comment:
The `Node` class constructor requires both `key` and `value` as arguments, but the current implementation only passes `key`. Ensure both `key` and `value` are passed to the `Node` constructor.